caikit/runtime/servicers/global_train_servicer.py

```diff
@@ -209,26 +225,37 @@ def run_training_job(
             **build_caikit_library_request_dict(request, model.train),
         }
 
-        # If running with a subprocess, set the target and args accordingly
-        target = (
-            self._train_and_save_model_subproc
-            if self.use_subprocess
-            else self._train_and_save_model
-        )
+        # If running with a subprocess, set the target, events and args accordingly
+        if self.use_subprocess:
+            event = multiprocessing.Event()
+            target = SubProcessTrainSaveExecutor(event)
+        else:
+            event = threading.Event()
+            target = LocalTrainSaveExecutor(event)
+
         log.debug2(
             "Training with %s",
             "SUBPROCESS" if self.use_subprocess else "MAIN PROCESS",
         )
 
         # start training asynchronously
         thread_future = self.run_async(
-            runnable_func=target,
+            runnable_executor=target,
             kwargs=kwargs,
             model_name=model_name,
             model_path=model_path,
         )
+
         self.training_map[training_id] = thread_future
 
+        # Add callback to register cancellation of training
+        def cancel_future(*args, **kwargs):
```

Comment:
I think it's a little misleading to call this `cancel_future` since it will _always_ be called on termination. It would be good to make the semantics clear here that this is a termination callback that will perform cancelation if termination happens in an abnormal way